app/controllers/members_controller.rb

```diff
@@ -24,18 +26,34 @@ def create
     end
   end
 
-  def edit
+  def verify_email
+    @member = Member.find_by_email_confirmation_token(params[:id])
+    if @member
+      @member.activate!
+      MemberMailer.welcome_email(@member).deliver_now
+      flash[:success] = "Welcome to the S'Up! Your email has been confirmed."
+      redirect_to action: :index
+    else
+      flash[:error] = 'Invalid token'
+      redirect_to action: :index
+    end
+  end
+
+  def getter
     @member = Member.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    redirect_to '/404.html'
```

Comment:
See http://stackoverflow.com/questions/2385799/how-to-redirect-to-a-404-in-rails for how to redirect to a 404 in Rails. Also needs a test.